### ISSUE TITLE
How to run pm2 command on AWS Elastic Beanstalk?

### ISSUE THREAD

**Issue by iForests** [Unitech/pm2#2193]
Hi. I followed [Using PM2 in Cloud Providers](http://pm2.keymetrics.io/docs/usage/use-pm2-with-cloud-providers/) and deployed to AWS Elastic Beanstalk successfully. But my problem is, when I run command like `pm2 monit`, bash tell me that `pm2: command not found`. I've tried something like the following but in vain:

- First, export the path of node:

```
PATH=$PATH:`ls -td /opt/elasticbeanstalk/node-install/node-* | head -1`/bin
```

- Then run `pm2 list` in this folder, but got an empty list:

```
ec2-user@ip-10-153-160-238 ~]$ /var/app/current/node_modules/pm2/bin/pm2 list
┌──────────┬────┬──────┬─────┬────────┬─────────┬────────┬────────┬──────────┐
│ App name │ id │ mode │ pid │ status │ restart │ uptime │ memory │ watching │
└──────────┴────┴──────┴─────┴────────┴─────────┴────────┴────────┴──────────┘
 Use `pm2 show <id|name>` to get more details about an app
```

What should I do to make it work? Thanks.

**Comment by walston** [comment]
try `node_modules/.bin/pm2 monit` instead of `pm2 monit`

looks like you didn't install your pm2 globally, which is fine, but your bash doesn't know to look into `node_modules/bin`

**Comment by iForests** [comment]
I've tried `/var/app/current/node_modules/pm2/bin/pm2 list` but got an empty list.

**Comment by antonsamper** [comment]
I had the same issue and it looks like the problem is to do with how PM2 starts up. My problem was that i was using the AWS EB interface to set the node command e.g. `npm run serve` which in turn would run `pm2 start...`. After some debugging, I realised that `pm2` was being started by the `nodejs` user instead of `ec2-user` and thats the reason why i was getting an empty list, because the user was wrong.

In order to see the apps running I had to run the following command:
`sudo -u nodejs pm2 list`

**Comment by iForests** [comment]
`sudo -u nodejs pm2 list` Works for me too. Thank you @antonsamper!

**Comment by ArMouReR** [comment]
I know that the thread already closed, but may this will save time for the next one that will google it:
`sudo -u nodejs /opt/elasticbeanstalk/node-install/node-v6.10.0-linux-x64/bin/node  /var/app/current/node_modules/.bin/pm2 monit`

Amazon Elastic Beanstalk environment, 64bit Amazon Linux 2016.09, pm2 installation according to this [guide](http://pm2.keymetrics.io/docs/tutorials/use-pm2-with-aws-elastic-beanstalk/), e.g.
`"scripts": {
  "start": "node ./node_modules/.bin/pm2 start app.js --name yourApp",
  "poststart":  "node ./node_modules/.bin/pm2 logs"
}`

**Comment by ahelord** [comment]
@antonsamper What was the command then to start pm2? I understand then that it would be this

> sudo -u nodejs node ./node_modules/.bin/pm2 start app.js

**Comment by ArMouReR** [comment]
You should make sure that node is in your PATH.
If not you should specify the full path, something like /opt/elasticbeanstalk/node-install/node-v6.10.0-linux-x64/bin/node

**Comment by antonsamper** [comment]
@ArMouReR is right, it sounds like a PATH issue.

I get around this PATH issue by binding the PATH straight away with `.ebextensions` where i have the following lines:

```yaml
container_commands:
  01_pm2_install:
      command: "if [ ! -e /bin/pm2 ]; then npm install pm2@2.0.18 -g; fi"
  02_pm2_binary:
    command: "ln -sf `ls -td /opt/elasticbeanstalk/node-install/node-* | head -1`/bin/pm2 /bin/pm2"```

**Comment by love8587** [comment]
This is what I used 

```
sudo ln -sf `ls -td /opt/elasticbeanstalk/node-install/node-* | head -1`/bin/node /bin/node
sudo -u nodejs node /var/app/current/node_modules/.bin/pm2 list
```

**Comment by byeonghun-lee** [comment]
I'll write down how I did it for those who see it later.
I used Node.js 14 running on 64bit Amazon Linux 2 in beanstalk. 
At first it worked the way @ArMouReR  did, but after updating to Amazon Linux 2 it didn't work.
I got an error saying "nodejs unknown user" and I got the user list through `cat /etc/passwd.`

I found the user name webapp in the list, adjusted the version to me, ran it, and checked the correct operation.

```
sudo -u webapp /opt/elasticbeanstalk/node-install/node-v14.19.1-linux-x64/bin/node /var/app/current/node_modules/.bin/pm2 monit
or
 sudo -u webapp /var/app/current/node_modules/.bin/pm2 monit
```
Thank you @ArMouReR
